src/backend/aspen/api/utils/sample.py

```diff
@@ -32,10 +35,17 @@ async def samples_by_identifiers(
         .outerjoin(public_samples_query, Sample.id == public_samples_query.c.id)  # type: ignore
         .outerjoin(private_samples_query, Sample.id == private_samples_query.c.id)  # type: ignore
         .where(
-            or_(
-                public_samples_query.c.id != None,  # noqa: E711
-                private_samples_query.c.id != None,  # noqa: E711
-            )
+            and_(
+                or_(
+                    # TODO - DECOVIDIFY - remove the None check!
+                    Sample.pathogen == pathogen,  # noqa: E711
+                    Sample.pathogen_id == None,  # noqa: E711
+                ),
+                or_(
+                    public_samples_query.c.id != None,  # noqa: E711
+                    private_samples_query.c.id != None,  # noqa: E711
+                ),
+            ),
```

Comment:
This adds a filter to select samples for a given type of pathogen (what we want once all samples have pathogens) OR no pathogen (legacy behavior). We need to leave this `== None` thing in place until all newly created samples have pathogen information attached, *and* we migrate all old samples to have pathogens.